odoo/addons/base/ir/ir_model.py

```diff
@@ -829,7 +841,7 @@ def _reflect_field(self, field):
             "SELECT f.*, d.module, d.id as xmlid_id, d.name as xmlid "
             "FROM ir_model_fields f LEFT JOIN ir_model_data d "
             "ON f.id=d.res_id and d.model='ir.model.fields' WHERE f.model=%s",
-            (self._name,))
+            (field.model_name,))
```

Comment:
@pedrobaeza I think this is the source of the problems you observed with fields not being moved to their new home module properly. My mistake when porting the framework to 11.0. Sorry for the inconvenience!

Reply:
No worries. The important thing is that it gets fixed in the best possible way.